### ISSUE TITLE
polymer build: CSS imports end up in wrong component with --bundle

### ISSUE THREAD

**Issue by simonfuhrmann** [Polymer/polymer-cli#917]
### Description
When using `polymer build --bundle` to bundle HTML and CSS into a single file, CSS rules that are imported using `<link rel="import" type="css" href="diy-content.css">` can (sometimes) end up in the wrong component.

### Versions & Environment
`polymer --version` 1.5.7
`node --version` v6.11.2
`uname` Linux 4.10.8-100.fc24.x86_64 x86_64 GNU/Linux

#### Steps to Reproduce
- Create directory `test/`.
- In `test/`, install Polymer: `bower install --save Polymer/polymer`
- Create the files below in `test/`
- Run `polymer build --bundle`

diy-app.css
```
div {
  color: red;
}
```

diy-content.css
```
div {
  font-weight: bold;
}
```

diy-app.html
```
<link rel="import" href="bower_components/polymer/polymer-element.html">
<link rel="import" href="diy-content.html">

<dom-module id="diy-app">
  <link rel="import" type="css" href="diy-app.css">
  <template>
    <div>DIY APP</div>
    <diy-content></diy-content>
  </template>
</dom-module>

<script>
class DiyApp extends Polymer.Element {
  static get is() {
    return "diy-app";
  }
}
customElements.define(DiyApp.is, DiyApp);
</script>
```

diy-content.html
```
<link rel="import" href="bower_components/polymer/polymer-element.html">

<dom-module id="diy-content">
  <link rel="import" type="css" href="diy-content.css">
  <template>
    <div>DIY CONTENT</div>
  </template>
</dom-module>

<script>
class DiyContent extends Polymer.Element {
  static get is() {
    return "diy-content";
  }
}
customElements.define(DiyContent.is, DiyContent);
</script>
```

index.html
```
<!DOCTYPE html>
<html lang="en">
  <head>
    <script src="bower_components/webcomponentsjs/webcomponents-lite.js"></script>
    <link rel="import" href="diy-app.html">
  </head>
  <body>
    <diy-app></diy-app>
  </body>
</html>
```

#### Expected Results
"DIY APP" should be in red, "DIY CONTENT" should be bold.

#### Actual Results
"DIY CONTENT" is both red and bold. The CSS styles for <diy-app> are inlined into the <diy-content> component.

**Comment by simonfuhrmann** [comment]
A new notes:
- When inlining the CSS, this issue does not occur (obviously).
- When removing the CSS import from `diy-content.html`, the issue does not occur.

**Comment by usergenic** [comment]
@simonfuhrmann I am a little confused.  The "actual results" you posted:
```html
<style>div {
  font-weight: bold;
}</style><style>div {
  color: red;
}</style>
```
appears to show inlined css?  Yet you said:

> When inlining the CSS, this issue does not occur (obviously)

Can you clarify the issue?

**Comment by simonfuhrmann** [comment]
The actual results I posted is the generated output from `polymer build --bundle`. It inlines the CSS (the whole point of bundle), but in the wrong component.

What I meant in my notes is: If you inline the CSS in the source files, and don't use an import, the issue does not occur. Makes sense?

**Comment by usergenic** [comment]
Okay.  Got it.  Thanks, that makes sense.  Definitely looks like a bug with the inlining then.  Will try to add a regression test for this and patch it up straight away.

As a workaround/change the Bundler does support inlining regular stylesheet links inside templates `<template><link rel="stylesheet" href="diy-app.css"></template>` which *should* not have this particular css import bug.

**Comment by usergenic** [comment]
@simonfuhrmann fixing in bundler https://github.com/Polymer/polymer-bundler/pull/609

**Comment by simonfuhrmann** [comment]
The title of your commit "Fixed deprecated CSS import links ..." leads me to believe that the type of import I am using is deprecated. Is that true? What is the non-deprecated alternative to import external per-component CSS styles?

Thanks for the quick fix!

**Comment by usergenic** [comment]
So in Polymer 1.0 `<link rel=import type=css href=...>` css imports were deprecated in favor of [style modules](https://www.polymer-project.org/1.0/docs/devguide/styling#style-modules).

However, if all you want to achieve is externalizing the css code for authoring purposes, you can use bundler to inline css like this:
```html
<template>
    <link rel=stylesheet href=diy-app.css>
    <div>DIY APP</div>
    <diy-content></diy-content>
</template>
```

You should still look at using style modules for *shared* external css, but if the css is specific to the element, this works great.  Note the `<link>` goes *inside* the `<template>` unlike the `<link rel=import type=css href=...>` css import examples you'll find from old (pre 1.0?) docs.

**Comment by simonfuhrmann** [comment]
Yes, all I want is to externalize the CSS, not share styles. I am aware of style modules, but I'll consider switching to `<link rel="stylesheet" ...>` for externalizing. Thanks!
